### DISCUSSION TITLE
Transpose by Semitone

### DISCUSSION THREAD

**Discussion by longlongso** [opensheetmusicdisplay/opensheetmusicdisplay#1249] [General]
There are some problems in semitone translation. When the score is not in C key, the conversion result is wrong. For non C key music score, the initial value of "openSheetMusicDisplay. Sheet. Transfer" should not be 0, and the original key should be considered before transposing. This is what we want in practical use.

**Comment by matt-uib** [comment]
The current function shall stay as you typically set a transpose value from the original key in a relative manner. e.g. -3 semitones on your keyboard...
But it makes also sense to add an absolute getter / setter function where C = 0 and e.g. D = 2 ... 
there the getter would give back 2 if the piece was originally in D.
and with the setter you could set it then to 7 which would be transposing to G then.

**Comment by sschmidTU** [comment]
Absolute transposition would also not make sense if you have a transposing instrument (e.g. Clarinet in A) among non-transposing instruments (e.g. piano) in your score.
But yes, we could offer helper functions for absolute transposition.
Meanwhile, you could add this function yourself by checking the current key and calculating the number of halftones to the target key.
